source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -367,6 +357,137 @@ string MdlShaderGenerator::getUpstreamResult(const ShaderInput* input, GenContex
+                                    return false; // no need to continue with this subgraph
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                // Check for transmission BSDF node.
+                if (node->hasClassification(ShaderNode::Classification::BSDF_T))
+                {
+                    // Check if IOR is connected.
+                    ShaderInput* ior = node->getInput("ior");
+                    ShaderOutput* source = ior ? ior->getConnection() : nullptr;
+                    if (source)
+                    {
+                        // Check if this is a graph interface connection.
+                        if (source->getNode() == g)
+                        {
+                            graphsWithIorUniform.insert(g);
+                            source->setFlag(ShaderPortFlagMdl::TRANSMISSION_IOR_DEPENDENCY, true);
+                            result = true;
+                        }
+                        else if (source->getNode()->hasClassification(ShaderNode::Classification::CONSTANT))
+                        {
+                            // If the connection is to a constant node we can
+                            // handled that here since it's just a uniform value.
+                            ShaderInput* value = source->getNode()->getInput(ValueElement::VALUE_ATTRIBUTE);
+                            if (value && value->getValue())
+                            {
+                                ior->setValue(value->getValue());
+                            }
+                            ior->breakConnection();
+                        }
+                        else
+                        {
+                            // If we get here we have to assume this is a varying connection
+                            // and we can break it immediately here.
+                            ior->breakConnection();
+                        }
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    // Disconnect any incomming connections to transmission IOR
+    // inside a graph.
+    void disconnectTransmissionIor(ShaderGraph* g)
+    {
+        for (ShaderNode* node : g->getNodes())
+        {
+            ShaderGraph* subgraph = node->getImplementation().getGraph();
+            if (subgraph && (subgraph->hasClassification(ShaderNode::Classification::SHADER) ||
+                             subgraph->hasClassification(ShaderNode::Classification::CLOSURE)))
+            {
+                disconnectTransmissionIor(subgraph);
+            }
+            else if (node->hasClassification(ShaderNode::Classification::BSDF_T))
+            {
+                ShaderInput* ior = node->getInput("ior");
+                if (ior)
+                {
+                    ior->breakConnection();
```

Comment:
Curious if you ever get here since you've done the traversal above and if you encounter one of these nodes then isn't the connection already broken ? Guess this is generic though to handle all cases.